### ISSUE TITLE
[React Floating menu] carry a11y updates from vanilla

### ISSUE THREAD

**Issue by elizabethsjudd** [carbon-design-system/carbon#3817]
a11y updates made to floating menu components that should be carried over to react

https://github.com/carbon-design-system/carbon/pull/3652
